test/Microsoft.TestTemplates.Acceptance.Tests/AcceptanceTests.cs

```diff
@@ -0,0 +1,151 @@
+
+using System.Collections.Immutable;
+
+namespace Microsoft.TestTemplates.Acceptance.Tests;
+
+[TestClass]
+public sealed partial class AcceptanceTests
+{
+    private static readonly ImmutableArray<string> SupportedTargetFrameworks =
+    [
+        "net8.0",
+        "net9.0",
+    ];
+    private static readonly (string ProjectTemplateName, string ItemTemplateName, string[] Languages)[] AvailableItemTemplates =
+    [
+        ("nunit", "nunit-test", Languages.All),
+    ];
+
+    private static readonly (string ProjectTemplateName, string[] Languages, bool RunDotnetTest)[] AvailableProjectTemplates =
+    [
+        ("mstest", Languages.All, true),
+        ("nunit", Languages.All, true),
+        ("xunit", Languages.All, true),
+        ("mstest-playwright", [Languages.CSharp], false),
+        ("nunit-playwright", [Languages.CSharp], false),
+    ];
+
+    [AssemblyInitialize]
+    public static void InstallTemplates(TestContext _)
+    {
+        foreach (var targetFramework in SupportedTargetFrameworks)
+        {
+            DotnetUtils.InvokeDotnetNewInstall(GetTestTemplatePath(targetFramework));
+        }
+    }
+
+    [AssemblyCleanup]
+    public static void UninstallTemplates()
+    {
+        foreach (var targetFramework in SupportedTargetFrameworks)
+        {
+            DotnetUtils.InvokeDotnetNewUninstall(GetTestTemplatePath(targetFramework));
+        }
+    }
+
+    [DataTestMethod]
+    [DynamicData(nameof(GetTemplateItemsToTest), DynamicDataSourceType.Method)]
+    public void ItemTemplate_CanBeInstalledAndTestArePassing(string targetFramework, string projectTemplate, string itemTemplate, string language)
+    {
+        string testProjectName = GenerateTestProjectName();
+        string outputDirectory = Path.Combine(Constants.ArtifactsTempDirectory, testProjectName);
+
+        // Create new test project: dotnet new <projectTemplate> -n <testProjectName> -f <targetFramework> -lang <language>
+        DotnetUtils.InvokeDotnetNew(projectTemplate, testProjectName, targetFramework, language, outputDirectory);
+
+        var itemName = "test";
+
+        // Add test item to test project: dotnet new <itemTemplate> -n <test> -lang <language> -o <testProjectName>
+        DotnetUtils.InvokeDotnetNew(itemTemplate, itemName, language: language, outputDirectory: outputDirectory);
+
+        if (language == Languages.FSharp)
+        {
+            // f# projects don't include all files by default, so the file is created
+            // but the project ignores it until you manually add it into the project
+            // in the right order
+            AddItemToFsproj(itemName, outputDirectory, testProjectName);
+        }
+
+        // Run tests: dotnet test <path>
+        var result = DotnetUtils.InvokeDotnetTest(outputDirectory);
+
+        // Verify the tests run as expected.
+        result.ValidateSummaryStatus(2, 0, 0);
+
+        Directory.Delete(outputDirectory, true);
```

Comment:
This is not guaranteed to execute. Test cleanup or using or finally?

Reply:
OR just put them in (the right) temp, and leave them, makes it much easier to debug locally.

Reply:
I thought about keeping the directory in case of error so we can "easily" investigate.